helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -170,6 +174,8 @@ public void close() {
 
     LOG.info("Finish computing new ideal states for resources: {}",
         resourceMap.keySet().toString());
+    // Register all metrics
+    _metricCollector.register();
```

Comment:
Considering future improvement, I mean constructing the WAGED rebalance only once, please register in the constructor. Or even earlier. You can pass the registered _metricCollector to the WAGED rebalancer so it does not need to register again and again.

Another question is that I assume metric data has been updated here, why we register after the data has been collected?